tauri/src/app.rs

```diff
@@ -62,6 +62,14 @@ impl<T: Serialize> From<T> for InvokeResponse {
   }
 }
 
+#[derive(Deserialize)]
+#[allow(missing_docs)]
+#[serde(tag = "cmd", rename_all = "camelCase")]
+pub struct DispatchInstructions {
```

Comment:
this forces the user to call the API with `{ type: 'myType', args: JSON.stringify({ my: 'args', like: 'this' })` right?

We can skip the JSON.stringify need by declaring the struct like this:

```rust
#[derive(Deserialize)]
pub struct DispatchInstructions {
  pub cmd: String,
  #[serde(flatten)]
  pub args: serde_json::Value,
}
```

Reply:
also, this struct could just be defined on the macro instead right? probably better

Reply:
Yeah it does require that. Would that move all other fields in the object into the `args` param then?
I also think we should change the `invoke` js function to function like `invoke('command_name', args)` instead of grouping the params and command name together. Especially if someone sorts their objects alphabetically, it could be confusing to have the command name stuck in with the args.

Reply:
yeah every field that's not named `cmd` would be part of the `args` value.

Reply:
yeah i agree with the invoke change, we can do it on the next PR.

Reply:
btw should we change cases here? `invoke('commandName')` to run the `fn command_name`? since the default on JS is camelCase, and on Rust it's snake_case

Reply:
I just added the flatten stuff :+1: